spec/requests/api/v1/invoices_request_spec.rb

```diff
@@ -16,22 +16,155 @@
-    expect(invoice).to have_key("created_at")
-    expect(invoice).to have_key("updated_at")
+    expect(invoice).not_to have_key("created_at")
+    expect(invoice).not_to have_key("updated_at")
     expect(invoice).to have_key("customer_id")
     expect(invoice).to have_key("merchant_id")
   end
 
   it "can show single invoice" do
     customer = create(:customer)
     merchant = create(:merchant)
-    id = create(:invoice, customer_id: customer.id, merchant_id: merchant.id).id
+    invoice = create(:invoice, customer_id: customer.id, merchant_id: merchant.id)
 
-    get "/api/v1/invoices/#{id}"
+    get "/api/v1/invoices/#{invoice.id}"
 
-    invoice = JSON.parse(response.body)
+    invoice_response = JSON.parse(response.body)
 
     expect(response).to be_success
-    expect(invoice["id"]).to eq(id)
+    expect(invoice_response["id"]).to eq(invoice.id)
+    expect(invoice_response["status"]).to eq(invoice.status)
+    expect(invoice_response["customer_id"]).to eq(customer.id)
+    expect(invoice_response["merchant_id"]).to eq(merchant.id)
+  end
+end
+
+describe "Invoice find, find all, and random" do
+  before(:each) do
+    @customer = create(:customer)
+    @merchant = create(:merchant)
+    @invoice = Invoice.create!(customer: @customer, merchant: @merchant,
+                              status: "shipped", created_at: "2012-03-06T16:54:31",
+                              updated_at: "2012-03-06T16:54:31")
+    @invoice2 = Invoice.create!(customer: @customer, merchant: @merchant,
+                              status: "shipped", created_at: "2012-03-06T16:54:31",
+                              updated_at: "2012-03-06T16:54:31")
+  end
+
+  it "can find single invoice by id" do
+    get "/api/v1/invoices/find?id=#{@invoice.id}"
+
+    invoice_response = JSON.parse(response.body)
+
+    expect(response).to be_success
+    expect(invoice_response["id"]).to eq(@invoice.id)
+  end
+
+  it "can find single invoice by status" do
+    get "/api/v1/invoices/find?status=shipped"
+
+    invoice_response = JSON.parse(response.body)
+
+    expect(response).to be_success
+    expect(invoice_response["status"]).to eq("shipped")
+  end
+
+  it "can find single invoice by customer_id" do
+    get "/api/v1/invoices/find?customer_id=#{@customer.id}"
+
+    invoice_response = JSON.parse(response.body)
+
+    expect(response).to be_success
+    expect(invoice_response["customer_id"]).to eq(@customer.id)
+  end
+
+  it "can find single invoice by merchant_id" do
+    get "/api/v1/invoices/find?merchant_id=#{@merchant.id}"
+
+    invoice_response = JSON.parse(response.body)
+
+    expect(response).to be_success
+    expect(invoice_response["merchant_id"]).to eq(@merchant.id)
+  end
+
+  it "can find single invoice by created_at" do
+    get "/api/v1/invoices/find?created_at=#{@invoice.created_at}"
+
+    invoice_response = JSON.parse(response.body)
+
+    expect(response).to be_success
+    expect(invoice_response["id"]).to eq(@invoice.id)
```

Comment:
I think it might be helpful in these tests to check for the search parameter in the response. Maybe an expect(invoice_response["created_at"] instead? With the arrays further down, maybe pull out the first object and check the params of that object.

Reply:
created_at will not be in the response because we are filtering it out.